src/reducers/vis-state-updaters.js

```diff
@@ -634,44 +634,125 @@ export const updateAnimationSpeedUpdater = (state, action) => ({
+ * Update animation config current time
  * @memberof visStateUpdaters
  * @param {Object} state `visState`
  * @param {Object} action action
- * @param {Number} action.value current time value of action
+ * @param {Number} action.speed current speed of animation
  * @returns {Object} nextState
  * @public
  *
  */
 
-export const playAnimationUpdater = (state, {value}) => ({
+export const playAnimationUpdater = (state, {speed}) => ({
   ...state,
   animationConfig: {
     ...state.animationConfig,
-    currentTime: value
+    currentTime: state.animationConfig.currentTime + speed
   }
 });
 
 /**
- * Enable animation domain with the min and max of timestamps from geojson
+ * Reset animation config current time to a specified value
  * @memberof visStateUpdaters
  * @param {Object} state `visState`
  * @param {Object} action action
+ * @param {Number} action.value the value current time will be set to
  * @returns {Object} nextState
  * @public
  *
  */
 
-export const enableLayerAnimationUpdater = (state, action) => {
-  const {oldLayer} = action;
-  const [minTs, maxTs] = getTimeAnimationDomain(oldLayer, state.datasets);
+export const resetAnimationUpdater = (state, {value}) => ({
+  ...state,
+  animationConfig: {
+    ...state.animationConfig,
+    currentTime: value
+  }
+});
+
+function updateAnimationDomain(state) {
+  // merge all animatable layer domain calculate the union
+  // state.layers.config.animation.domain
+  // take state.layer
+  const animatableLayers = state.layers.filter(l => l.config.animation.enabled);
+  // const domains = state.layers.map(l => l.config.animation.domain);
+
+  const minRange = [];
+  const maxRange = [];
+
+  animatableLayers.map(layer => {
+    const layerTimeRange = layer.config.animation.domain;
+
+    minRange.push(layerTimeRange[0]);
+    maxRange.push(layerTimeRange[1]);
+  });
+
+  const minTs = extent(minRange)[0];
+  const maxTs = extent(maxRange)[1];
+
   return {
     ...state,
     animationConfig: {
       ...state.animationConfig,
-      currentTime: minTs,
+      // of currentTime is in newDomain, don't change it
+      // if currentTime is not in newDomain, set it to min
+      currentTime: minTs, // Math.min(currentMin, minTs),
+      domain: [minTs, maxTs] // [Math.min(currentMin, minTs), Math.max(currentMax, maxTs)]
+    }
+  };
+}
+
+/**
+ * Set animation domain with the min and max of timestamps from geojson
+ * Enable multi-layer domain range
+ * @memberof visStateUpdaters
+ * @param {Object} state `visState`
+ * @param {Object} action action
+ *  @param {Object} action.oldLayer the layer object to be updated
+ * @returns {Object} nextState
+ * @public
+ *
+ */
+
+export const enableLayerAnimationUpdater = (state, oldLayer) => {
+  // calculate animation domain for the layer enabled
+  // layer.config.animation.domain
+  // update newLayer into the state.layers
+  // merge all animatable layer domain calculate the union
+  const [minTs, maxTs] = getTimeAnimationDomainForTripLayer(
```

Comment:
if you are not use `minTx` and `maxTs` separately, don't pull it out

`const domain = getTimeAnimationDomainForTripLayer`